### PR TITLE
Events

### PR DESCRIPTION
**Pull request by chrisheckler**
Events now have basic structure.
- Event Creation 
- Event Listing
- Voting : Votes limited to 1 per person., aggregation across users 
- Detail page for each event
- Update : Only owners can update events

ToDo
File uploads
styling

### CODE REVIEW

File: world_challenge_events/views.py [hunk 1]
```diff
@@ -1,16 +1,80 @@
...
+class EventCreateFormView(CreateView):
+    """This view handles app logic for event creation"""
+
+    model = Event
+    form_class = EventCreateForm
+    template_name = "world_challenge_events/event_create.html"
+
+    def post(self, request, *args, **kwargs):
+        form = self.form_class(request.POST)
+        if form.is_valid():
```
Comment (jamesbeedy) [inline world_challenge_events/views.py:40]: what happens if the form is *not* valid?

File: world_challenge_events/views.py [hunk 2]
```diff
@@ -1,16 +1,80 @@
...
+
+
+class EventDetailView(DetailView):
+    """Home page for individual events"""
+
+    model = Event
+
+    def post(self, request, *args, **kwargs):
+        event = Event.objects.get(pk=kwargs['pk'])
+        event.votes.up(request.user.id)
```
Comment (jamesbeedy) [inline world_challenge_events/views.py]: what happens if the user has already voted?

File: world_challenge_events/views.py [hunk 3]
```diff
@@ -1,16 +1,80 @@
...
+    template_name = 'world_challenge_events/event_update.html'
+
+    def dispatch(self, request, *args, **kwargs):
+        event = Event.objects.get(pk=kwargs['pk'])
+        if event.owner == self.request.user:
+            try:
+                return super(EventUpdateView, self).dispatch(request, *args, **kwargs)
+            except:
+
+                messages.add_message(request, messages.INFO, 'Event has been Updated!!')
```
Comment (jamesbeedy) [inline world_challenge_events/views.py:81]: where is the messages object initialized?

**Review by jamesbeedy — CHANGES_REQUESTED** [review]

File: world_challenge_events/views.py (hunk 3 above)
Comment (chrisheckler) [reply world_challenge_events/views.py:81]: On the base.html

File: world_challenge_events/views.py (hunk 2 above)
Comment (chrisheckler) [reply world_challenge_events/views.py]: Form message and redirect is initialized

**Review by jamesbeedy — APPROVED** [review]

**Pull request merged**